app/controllers/trips_controller.rb

```diff
@@ -1,17 +1,33 @@
 class TripsController < ApplicationController
-  before_action :set_trip, only: [:dashboard, :show, :edit, :update, :destroy]
+  before_action :set_trip, only: [:dashboard, :import, :show, :edit, :update, :destroy]
+  skip_before_action :verify_authenticity_token, only: [:import]
 
   # GET /trips
   # GET /trips.json
   def index
     authorize Trip
-    @trips = Trip.all
-  end
-
+    @trips = Trip.all end
   def dashboard
     authorize Trip
   end
 
+  def import
+    if request.post?
+      authorize Trip
+      @trip = Trip.find(params[:id])
```

Comment:
This line isn't necessary if `set_trip` is called for this action. You can see that you added it to the before_action on line 2, so `@trip` is already in scope here.